ledger/sandbox-classic/src/main/scala/platform/sandbox/stores/InMemoryActiveLedgerState.scala

```diff
@@ -26,18 +26,22 @@ private[sandbox] case class InMemoryActiveLedgerState(
     parties: Map[Party, PartyDetails],
 ) extends ActiveLedgerState[InMemoryActiveLedgerState] {
 
-  def isVisibleForDivulgees(contractId: ContractId, forParty: Party): Boolean =
+  def isVisibleForDivulgees(contractId: ContractId, forParties: Set[Party]): Boolean =
     activeContracts
       .get(contractId)
-      .exists(ac => ac.witnesses.contains(forParty) || ac.divulgences.contains(forParty))
+      .exists(ac =>
+        ac.witnesses.exists(forParties.contains) || ac.divulgences.exists {
+          case (party, _) => forParties.contains(party)
+      })
```

Comment:
This could now also be written as 
```suggestion
        forParties.exists(p => ac.witnesses.contains(p) || ac.divulgences.contains(p))
```